tests/models/idefics3/test_modeling_idefics3.py

```diff
@@ -0,0 +1,529 @@
+            output_embeds = model.get_output_embeddings()
+            self.assertEqual(output_embeds.weight.shape[0], model_vocab_size + 10)
+            # Check bias if present
+            if output_embeds.bias is not None:
+                self.assertEqual(output_embeds.bias.shape[0], model_vocab_size + 10)
+            # Check that the model can still do a forward pass successfully (every parameter should be resized)
+            model(**self._prepare_for_class(inputs_dict, model_class))
+
+            # Check that resizing the token embeddings with a smaller vocab size decreases the model's vocab size
+            model.resize_token_embeddings(model_vocab_size - 15)
+            self.assertEqual(model.config.text_config.vocab_size, model_vocab_size - 15)
+            # Check that it actually resizes the embeddings matrix
+            output_embeds = model.get_output_embeddings()
+            self.assertEqual(output_embeds.weight.shape[0], model_vocab_size - 15)
+            # Check bias if present
+            if output_embeds.bias is not None:
+                self.assertEqual(output_embeds.bias.shape[0], model_vocab_size - 15)
+
+            # Check that the model can still do a forward pass successfully (every parameter should be resized)
+            # Input ids should be clamped to the maximum size of the vocabulary - 1 and the image token should be the last token
+            inputs_dict["input_ids"].clamp_(max=model_vocab_size - 15 - 2)
+            n_images = self.model_tester.num_images * self.model_tester.seq_length
+            model.model.image_token_id = model_vocab_size - 15 - 1
+            inputs_dict["input_ids"][:, -n_images:] = model.model.image_token_id
+
+            # Check that the model can still do a forward pass successfully (every parameter should be resized)
+            model(**self._prepare_for_class(inputs_dict, model_class))
+
+
+@require_torch
+class Idefics3ForConditionalGenerationIntegrationTest(unittest.TestCase):
+    def setUp(self):
+        self.processor = AutoProcessor.from_pretrained("HuggingFaceM4/Idefics3-8B-Llama3")
+        self.image1 = Image.open(
+            BytesIO(
+                requests.get(
+                    "https://cdn.britannica.com/61/93061-050-99147DCE/Statue-of-Liberty-Island-New-York-Bay.jpg"
+                ).content
+            )
+        )
+        self.image2 = Image.open(
+            BytesIO(requests.get("https://cdn.britannica.com/59/94459-050-DBA42467/Skyline-Chicago.jpg").content)
+        )
+        self.image3 = Image.open(
+            BytesIO(
+                requests.get(
+                    "https://thumbs.dreamstime.com/b/golden-gate-bridge-san-francisco-purple-flowers-california-echium-candicans-36805947.jpg"
+                ).content
+            )
+        )
+
+    def tearDown(self):
+        gc.collect()
+        torch.cuda.empty_cache()
+
+    @slow
+    @unittest.skip("multi-gpu tests are disabled for now")
```

Comment:
These should only be skipped in the multi-gpu case and is a global problem not specific to this model. This will skip for single GPU too. 

It'd be better just to: 
* remove this skip
* before merge, final slow model tests run to confirm that the integration tests are correct
* do a final commit without a `[run-slow]` so you have a passing CI and can merge

Reply:
This test fails on single-gpu because the model takes too much RAM. That is why we need to run them on multi-gpu. Or is there something I'm not understanding from your message? I can't get a passing CI without multi-gpu support.

Reply:
The skip message doesn't reflect what is happening: this isn't skipping multi-gpu tests, it's skipping this test entirely, which means it doesn't run on in the single GPU or multi GPU case. 

In our offline conversation about multi-GPU tests, it was mentioned that it was too hard to debug as the runners were taking too long. This is a repo-wide issue and one which means slow model multi-gpu runs aren't a requirement for merging in. However, this is independent of the single GPU runs or if the model tests are failing in general. 

As a general note - skip messages should provide a clear reason, stating why the tests are being skipped and ideally link to a related github issue which can be tracked and ensure that the tests are restarted once resolved. At the moment, someone reading the code wouldn't know why this is happening or if the tests can be restarted.